diode-react/src/main/scala/diode/react/ReactConnector.scala

```diff
@@ -116,18 +116,19 @@ trait ReactConnector[M <: AnyRef] { circuit: Circuit[M] =>
       }
 
       private def changeHandler(cursor: ModelRO[S]): Unit = {
-        val isMounted = t.isMounted
+        val isMounted = t.isMounted.map(_.getOrElse(false))
```

Comment:
... in the other hand, I find it a bit redundant to have an update check in that method being invoked from the circuit subscription callback and then one in `shouldComponentUpdatePure` etc.

In my opinion the _update check_ should happen in a `CallbackTo[Boolean]` method that is bound in the `shouldComponentUpdate` and the subscription callback from the circuit should invoke `setState(...)` without any compassion. I find current logic a bit "split into two", which makes it a bit difficult to reason about.